### PR TITLE
fix coqdep not recognizing ignore import_categories syntax

### LINKED ISSUE #16673: coqdep doesn't recognize import_categories
#### Description of the problem

import_categories feature works perfectly fine in coqtop, but produces an error message while using coqdep.

Example:
```
echo "Require Import -(notations) stdpp.base." > test.v
coqdep -R . Top ./test.v
```

Error message:
`*** Error: File "test.v",characters 15-16: Syntax error`

#### Coq Version

The Coq Proof Assistant, version 8.16.0
compiled with OCaml 4.12.1

Operating system:
Linux, NixOS

### PR DESCRIPTION
**Pull request by Kaptch**
Fixes #16673 

- [x ] Added / updated **test-suite**. (a few tests for different cases of ignore import_categories modifiers)

### CODE REVIEW

File: tools/coqdep/lib/lexer.mll [hunk 1]
```diff
@@ -120,14 +120,10 @@ and extra_dep_rule from = parse
 and require_modifiers from = parse
   | "(*"
       { comment lexbuf; require_modifiers from lexbuf }
-  | "Import" space+
+  | "Import" space*
```
Comment (Alizter) [inline tools/coqdep/lib/lexer.mll:123]: I'm not sure about allowing no spaces between the `Import` and `-`. @SkySkimmer is this something we want to do?

Comment (SkySkimmer) [reply tools/coqdep/lib/lexer.mll:123]: That's how Coq syntax works

**Comment by Alizter** [comment]
I thought the `request: full CI` label actually did something.

@coqbot run full CI

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
merging soon

**Comment by Alizter** [comment]
@coqbot merge now

**Comment by Alizter** [comment]
Thanks!

**Pull request merged**